cmd/crates/soroban-spec-typescript/ts-tests/src/test.ts

```diff
@@ -0,0 +1,188 @@
+import test from 'ava'
+import { Contract, Ok, Err, networks } from '../../fixtures/test_custom_types/dist/esm/index.js'
+
+const rpcUrl = 'https://rpc-futurenet.stellar.org'
+const publicKey = 'GCBVOLOM32I7OD5TWZQCIXCXML3TK56MDY7ZMTAILIBQHHKPCVU42XYW'
+
+const contract = new Contract({
+  ...networks.futurenet,
+  rpcUrl,
+  wallet: {
+    isConnected: () => Promise.resolve(true),
+    isAllowed: () => Promise.resolve(true),
+    getUserInfo: () => Promise.resolve({ publicKey }),
+    signTransaction: async (tx: string, opts?: {
+      network?: string,
+      networkPassphrase?: string,
+      accountToSign?: string,
+    }) => tx,
+  },
+})
+
+test('hello', async t => {
+  t.is(await contract.hello({ hello: 'tests' }), 'tests')
+})
+
+test('woid', async t => {
+  t.is(await contract.woid(), undefined)
+})
+
+test('u32_fail_on_even', async t => {
+  t.deepEqual(await contract.u32FailOnEven({ u32_: 1 }), new Ok(1))
+  t.deepEqual(await contract.u32FailOnEven({ u32_: 0 }), new Err({ message: "Please provide an odd number" }))
+})
+
+test('u32', async t => {
+  t.is(await contract.u32({ u32_: 1 }), 1)
+})
+
+test('i32', async t => {
+  t.is(await contract.i32({ i32_: 1 }), 1)
+})
+
+test('i64', async t => {
+  t.is(await contract.i64({ i64_: 1n }), 1n)
+})
+
+test("strukt_hel", async (t) => {
```

Comment:
`struct` isn't reserved in JS, so why are we doing this weird mangling?

Reply:
It's the name of the functions in the contract.